problem3/problem3.rb

```diff
@@ -0,0 +1,31 @@
+class Problem3
+	attr_accessor :array
+
+	def  initialize(*args)
+		@array = args;
+	end
+
+	def DublicateNumbers
+		@array.concat Array.new(@array.size,0);
+		for i in 0..(@array.size/2-1)
+			if @array[@array.size/2+@array[i]-1] == 0
+				@array[@array.size/2+@array[i]-1] = -1;
+			elsif @array[@array.size/2+@array[i]-1] == -1
+				@array[@array.size/2+@array[i]-1]=1;
+			end
+		end
+		@array.shift(@array.size/2)
+		@array.push('-chuck-norris-')
+		for i in 0 .. @array.size-1
+			if(@array[i]==1)
+				array.push(i+1);
+			end
+		end
+		for i in 0.. @array.size-1
+			if @array[i] == '-chuck-norris-'
+				@array.shift(i+1);
+				break;
+			end
+		end
```

Comment:
Interesting solution, but we need a boolean as a result. The `Array.new` is new data structure. You could have filled the array another way to avoid instantiating a new array.